gateway.go

```diff
@@ -298,19 +315,19 @@ func (gw *Gateway) toWildcardQName(qName, zone string) string {
 
 // Gets the set of addresses associated with the first set of index keys
 // that is in the indexer.
-func (gw *Gateway) getMatchingAddresses(indexKeySets [][]string) []netip.Addr {
+func (gw *Gateway) getMatchingAddresses(indexKeySets [][]string) ([]netip.Addr, []string) {
 	// Iterate over supported resources and lookup DNS queries
 	// Stop once we've found at least one match
 	for _, indexKeys := range indexKeySets {
 		for _, resource := range gw.Resources {
-			addrs := resource.lookup(indexKeys)
-			if len(addrs) > 0 {
-				return addrs
+			addrs, txts := resource.lookup(indexKeys)
```

Comment:
txts should be any `non-ip`, not sure of a name we could use

Reply:
How will you detect what is a TXT record and what is a CNAME record if everything is in the same txts slice ? I mean nothing forbids to publish a domain name in a TXT record or even an IP address.